File: filter/filter/lib/filter_py/MovAvgFIlter.py
class MovAvgFilter:
    def __init__(self, window_size):
        self.window_size = window_size
        self.buffer = []

    def update(self, value):
        self.buffer.append(value)
        if len(self.buffer) > self.window_size:
            self.buffer.pop(0)

    def get_average(self):
        if len(self.buffer) == 0:
            return None
        return sum(self.buffer) / len(self.buffer)

File: filter/filter/lib/filter/MovAvgFilter.cpp
#include <filter/MovAvgFilter.h>

MovAvgFilter::MovAvgFilter(int window_size, double average_value)
    : window_size_(window_size), average_value_(average_value)
{
    for (int i = 0; i < window_size_; i++) {
        data_queue_.push(average_value_);
    }
}

double MovAvgFilter::GetEstimatedValue() const
{
    return average_value_;
}

double MovAvgFilter::Update(double input)
{
    data_queue_.push(input);
    average_value_ += (input - data_queue_.front()) / window_size_;
    data_queue_.pop();
    return average_value_;
}
